src/core/service.c

```diff
@@ -1532,6 +1532,35 @@ static int service_spawn_internal(
                 log_unit_debug(UNIT(s), "Passing %zu fds to service", exec_params.n_socket_fds + exec_params.n_storage_fds);
         }
 
+        /* These options require PrivateUsers= when used in user units, as we need to be in a user namespace
+         * to have permission to enable them when not running as root. */
+        if (!s->exec_context.private_users && !MANAGER_IS_SYSTEM(UNIT(s)->manager))
+                s->exec_context.private_users =
+                                s->exec_context.private_tmp ||
+                                s->exec_context.private_devices ||
+                                s->exec_context.private_network ||
+                                s->exec_context.private_ipc ||
+                                s->exec_context.private_mounts ||
+                                s->exec_context.mount_apivfs ||
+                                s->exec_context.dynamic_user ||
+                                s->exec_context.n_bind_mounts > 0 ||
+                                s->exec_context.n_temporary_filesystems > 0 ||
+                                s->exec_context.root_directory ||
+                                s->exec_context.extension_directories ||
```

Comment:
Up until this one the unit layout changes, so definitely should be enabled. For the ones below it only affects what they can touch though, so less sure.

Reply:
yeah, everything after this should be dropped.